core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -197,6 +167,48 @@ public void onError(Status error) {
     });
   }
 
+  @VisibleForTesting
+  static NameResolver getNameResolver(String target, NameResolver.Factory nameResolverFactory,
+      Attributes nameResolverParams) {
+    // Finding a NameResolver. Try using the target string as the URI. If that fails, try prepending
+    // "dns:///".
+    URI targetUri = null;
+    StringBuilder uriSyntaxErrors = new StringBuilder();
+    try {
+      targetUri = new URI(target);
+      // For "localhost:8080" this would likely return null, because "localhost" is parsed as the
```

Comment:
Please make the comment more accurate. "this would likely return null" -> "nameResolverFactory.newNameResolver() below would likely return null"